### PR TITLE
Refactor conditions logic

### PR DESCRIPTION
**Pull request by alexander-demichev**
Refactor conditions logic and make it more simple. PR also contains a commit that removes double imports.

### CODE REVIEW

File: pkg/actuators/machine/utils.go [hunk 1]
```diff
@@ -413,3 +351,20 @@ func extractNodeAddresses(instance *ec2.Instance) ([]corev1.NodeAddress, error)
...
+		Status:  corev1.ConditionTrue,
+		Reason:  providerconfigv1.MachineCreationSucceeded,
+		Message: "Machine successfully created",
+	}
+}
+
+func conditionFailed() providerconfigv1.AWSMachineProviderCondition {
+	return providerconfigv1.AWSMachineProviderCondition{
+		Type:   providerconfigv1.MachineCreation,
+		Status: corev1.ConditionTrue,
```
Comment (JoelSpeed) [inline pkg/actuators/machine/utils.go]: Should be false?
```suggestion
		Status: corev1.ConditionFalse,
```

File: pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go [hunk 2]
```diff
@@ -60,6 +60,16 @@ const (
 	MachineCreation AWSMachineProviderConditionType = "MachineCreation"
 )
 
+// AWSMachineProviderConditionReason is reason for the condition's last transition
+type AWSMachineProviderConditionReason string
+
+const (
+	// MachineCreationSucceeded indicates machine creation success
```
Comment (JoelSpeed) [inline pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go]: Nit: should have period at end
```suggestion
	// MachineCreationSucceeded indicates machine creation success.
```

File: pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go [hunk 3]
```diff
@@ -60,6 +60,16 @@ const (
 	MachineCreation AWSMachineProviderConditionType = "MachineCreation"
 )
 
+// AWSMachineProviderConditionReason is reason for the condition's last transition
```
Comment (JoelSpeed) [inline pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go]: Nit: should have period at end
```suggestion
// AWSMachineProviderConditionReason is reason for the condition's last transition.
```

File: pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go [hunk 4]
```diff
@@ -60,6 +60,16 @@ const (
...
+type AWSMachineProviderConditionReason string
+
+const (
+	// MachineCreationSucceeded indicates machine creation success
+	MachineCreationSucceeded AWSMachineProviderConditionReason = "MachineCreationSucceeded"
+	// MachineCreationFailed indicates machine creation fail
```
Comment (JoelSpeed) [inline pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go]: Nit: consistency
```suggestion
	// MachineCreationFailed indicates machine creation failure.
```

**Comment by alexander-demichev** [comment]
@JoelSpeed fixed

**Pull request merged**